script/stop-update.lua

```diff
@@ -182,16 +182,60 @@ function UpdateStop(stopID)
   -- .." requestThreshold:"..requestThreshold.." requestPriority:"..requestPriority.." noWarnings:"..tostring(noWarnings)
   -- .." provideThreshold:"..provideThreshold.." providePriority:"..providePriority.." lockedSlots:"..lockedSlots)
 
-  -- skip duplicated names on non depots
-  if #global.TrainStopNames[stop.entity.backer_name] ~= 1 and not isDepot then
-    stop.errorCode = 2
-    stop.activeDeliveries = {}
-    if stop.parkedTrainID and global.Dispatcher.availableTrains[stop.parkedTrainID] then
-      remove_available_train(stop.parkedTrainID)
+  -- store depot signal now because it's needed in check below
+  stop.isDepot = isDepot
+  
+  -- non depots may only have multiple stops with the same name if they are wired together
+  -- This allows building stations with multiple stops explicitly while detecting accidentally
+  -- reusing a name as might easily happen when blueprinting.
+  -- Depot and non-depot stops may not be mixed.
+  if #global.TrainStopNames[stop.entity.backer_name] ~= 1 then
+    local green, red
+    local errors = 0
+    for _, stopID in ipairs(global.TrainStopNames[stop.entity.backer_name]) do
+       local stop = global.LogisticTrainStops[stopID]
+       if stop.isDepot ~= isDepot then
+          errors = 3
+          break
+       end
+       local stop_green = stop.entity.get_circuit_network(defines.wire_type.green)
+       if stop_green then
+          stop_green = stop_green.network_id
+          if green and green ~= stop_green then
+             errors = bit32.bor(errors, 1)
+          end
+          green = stop_green
+       elseif not isDepot then
+          errors = bit32.bor(errors, 1)
+       end

```

Comment:
Did you benchmark if grabbing signals separately is better than grabbing all, and parsing them from a local table?
the optimized flow path should be:
- grab all signals
- pass error checks
- add signals readouts to global tables

Having slower performance on error neglect able and acceptable for depots.

Reply:
The new code now grabs the network_id for each stop only once. The drawback is that it takes 2 loops through all stops now.